File: 07/advent.jl
function parse(rawentry)
  fields = split.(split(rawentry), ":")
  dict = Dict()

  for field in fields
    dict[field[1]] = field[2]
  end

  dict
end

function validate(entry)
  haskey(entry, "byr") &&
  haskey(entry, "iyr") &&
  haskey(entry, "eyr") &&
  haskey(entry, "hgt") &&
  haskey(entry, "hcl") &&
  haskey(entry, "ecl") &&
  haskey(entry, "pid")
end

input = open(f -> read(f, String), "./input.txt")
entries = split(input, "\n\n")
parsed = parse.(entries)
valid = filter(x -> x, validate.(parsed))

println(length(valid))

File: 11/advent.py
file = open("./input.txt", "r")
contents = file.read()
entries = map(lambda entry: entry.split("\n"), contents.split("\n\n"))
entries = map(lambda entry: { letter for line in entry for letter in line}, entries)
counts = map(lambda entry: len(entry), entries)

print(sum(counts))
